huaweicloud/lb_v2_shared.go

```diff
@@ -92,6 +93,46 @@ func waitForLBV2LoadBalancer(networkingClient *golangsdk.ServiceClient, id strin
 	return nil
 }
 
+// for v2 api
+func waitForLBV2LoadBalancer_v2(networkingClient *golangsdk.ServiceClient, id string, target string, pending []string, timeout time.Duration) error {
+	log.Printf("[DEBUG] Waiting for loadbalancer %s to become %s.", id, target)
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{target},
+		Pending:    pending,
+		Refresh:    resourceLBV2LoadBalancerRefreshFunc_v2(networkingClient, id),
+		Timeout:    timeout,
+		Delay:      5 * time.Second,
+		MinTimeout: 1 * time.Second,
+	}
+
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			switch target {
+			case "DELETED":
+				return nil
+			default:
+				return fmt.Errorf("Error: loadbalancer %s not found: %s", id, err)
+			}
+		}
+		return fmt.Errorf("Error waiting for loadbalancer %s to become %s: %s", id, target, err)
+	}
+
+	return nil
+}
+
+func resourceLBV2LoadBalancerRefreshFunc_v2(networkingClient *golangsdk.ServiceClient, id string) resource.StateRefreshFunc {
```

Comment:
You can control the maximum number of characters in a single line (120) by wrapping lines or reduce the name description of this function.

Reply:
fixed